triage/2021-03-23.md

```diff
@@ -0,0 +1,66 @@
+
+2 Regressions, 5 Improvements, 3 Mixed
+1 of them in rollups
+
+#### Regressions
+
+Implement (but don't use) valtree and refactor in preparation of use [#82936](https://github.com/rust-lang/rust/issues/82936)
+- Moderate regression in [instruction counts](https://perf.rust-lang.org/compare.html?start=f5d8117c338a788bd24abec733fd143dfceb25a0&end=e655fb62216b6ba64a094b30f116d7988d19322d&stat=instructions:u) (up to 2.1% on `full` builds of `ctfe-stress-4-opt`)
+- Purely an addition of unused code (for a future feature). It is possible that this changed some inlining behavior, but the benchmark in question is susceptible to high variance though it seemed to impact full builds and not incremental builds.
+- The query impacted is `eval_to_allocation_raw` which is what ctfe stresses, so we'll [look into it](https://github.com/rust-lang/rust/pull/82936#issuecomment-805029518).
+
+Use TrustedRandomAccess for in-place iterators where possible [#79846](https://github.com/rust-lang/rust/issues/79846)
+- Moderate regression in [instruction counts](https://perf.rust-lang.org/compare.html?start=97663b6690689379aa0493deb494dfe14627c46b&end=35385770ae1ea86a911cc44ac43f856831e44b26&stat=instructions:u) (up to 1.5% on `full` builds of `deep-vector-debug`)
+- This is a change in the standard library that seems to only impact one benchmark: `deep-vector-debug` full compilation. It looks to be impacting `typeck`, but I'm not sure why this would be. It's also possible that it's noise.
+
+#### Improvements
+
+Add a check for ASCII characters in to_upper and to_lower [#81358](https://github.com/rust-lang/rust/issues/81358)
+- Moderate improvement in [instruction counts](https://perf.rust-lang.org/compare.html?start=2c7490379d7e8854192c176039cfabf6acefe7ef&end=0ce0fedb67fa66d50aa819ef8b12f1d89eb22d7d&stat=instructions:u) (up to -2.7% on `incr-unchanged` builds of `match-stress-enum-check`)
+
+ast/hir: Rename field-related structures [#83188](https://github.com/rust-lang/rust/issues/83188)
+- Moderate improvement in [instruction counts](https://perf.rust-lang.org/compare.html?start=04ae50179a802d1d18bb780baa85f55864f56616&end=b4adc21c4fa245994b4936df5b4f7d94ca633c5d&stat=instructions:u) (up to -1.7% on `incr-unchanged` builds of `deep-vector-check`)
+
+Revert performance-sensitive change in #82436 [#83293](https://github.com/rust-lang/rust/issues/83293)
+- Very large improvement in [instruction counts](https://perf.rust-lang.org/compare.html?start=6bfbf0c33a86707cedd02ca985285191282a80b3&end=cd82e451a88ac8b15894af498cdc49e1fe8caa21&stat=instructions:u) (up to -10.6% on `incr-full` builds of `packed-simd-check`)
+
+Rollup of 9 pull requests [#83360](https://github.com/rust-lang/rust/issues/83360)
+- Moderate improvement in [instruction counts](https://perf.rust-lang.org/compare.html?start=35385770ae1ea86a911cc44ac43f856831e44b26&end=142c831861ba5a995fd9de99198e7f6074b6b400&stat=instructions:u) (up to -1.5% on `full` builds of `deep-vector-debug`)
```

Comment:
... this one.

So it could be noise or one of the changes perturbing a thing and another change undoing it in a different way. I don't see a direct relation to the changes though (other than involving `Vec`) since the benchmark in question only constructs a vec without involving iterators.